test/distributed/org/apache/cassandra/distributed/test/ForceRepairTest.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.distributed.test;
+
+import java.io.IOException;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+import com.google.common.collect.ImmutableMap;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Test;
+
+import org.apache.cassandra.distributed.Cluster;
+import org.apache.cassandra.distributed.api.ICluster;
+import org.apache.cassandra.distributed.api.IInstanceConfig;
+import org.apache.cassandra.distributed.api.IInvokableInstance;
+
+import static org.apache.cassandra.distributed.api.Feature.GOSSIP;
+import static org.apache.cassandra.distributed.api.Feature.NETWORK;
+
+public class ForceRepairTest extends TestBaseImpl
+{
+    private Cluster cluster;
+
+    private static Cluster create(Consumer<IInstanceConfig> configModifier, int clusterSize, int replicationFactor) throws IOException
+    {
+        configModifier = configModifier.andThen(
+        config -> config.set("hinted_handoff_enabled", false)
+                        .set("commitlog_sync_batch_window_in_ms", 5)
+                        .with(NETWORK)
+                        .with(GOSSIP)
+        );
+
+        return init(Cluster.build().withNodes(clusterSize).withConfig(configModifier).start(), replicationFactor);
+    }
+
+    void forceRepair(ICluster<IInvokableInstance> cluster, Integer... downClusterIds) throws Exception
+    {
+        RepairTest.populate(cluster, "{'enabled': false}");
+        Stream.of(downClusterIds).forEach(id -> cluster.get(id).shutdown());
+        RepairTest.repair(cluster, ImmutableMap.of("forceRepair", "true"));
+    }
+
+    @After
+    public void closeCluster()
+    {
+        if (cluster != null)
+            cluster.close();
+    }
+
+    @Test
+    public void testForcedNormalRepairWithOneNodeDown() throws Exception
+    {
+        cluster = create(config -> {}, 4, 3);
```

Comment:
@yifan-c Can we simplify this test to use 3 nodes and an RF of 2? (i.e. We'd shut down node 2, but nodes 3 and 1 should constitute a full replica set.) If so, it seems like we could just add `testForcedNormalRepairWithOneNodeDown()` to `RepairTest` and avoid some of the duplication that exists between it and `ForceRepairTest`.

If we can't make that simplification, there is probably still room to create something like a `RepairTestBase` that can house some of the common elements.

Reply:
3/2 is good. Thanks for that. 

Finally remember the reason why having a separate test class. 
`RepairTest` share the same `cluster` across all tests. However, the `ForceRepairTest` need to shutdown a node during test. For clarity, I created a dedicated class.

Reply:
@yifan-c That's fair. We can either just make sure that node restarts before other tests start, or go with the `RepairTestBase`/`AbstractRepairTest` idea?

Reply:
Pushed a new commit to address it.